splunk-otel-android/src/test/java/com/splunk/rum/ActivityTracerCacheTest.java

```diff
@@ -0,0 +1,138 @@
+        assertSame(activityTracer, result);
+        verify(activityTracer).addEvent("beep");
+        verifyNoMoreInteractions(tracerCreator);
+    }
+
+    @Test
+    void addEventExistingActivity() {
+        when(tracerCreator.apply(activity)).thenReturn(activityTracer);
+        when(activityTracer.addEvent(anyString())).thenReturn(activityTracer);
+
+        ActivityTracerCache underTest = new ActivityTracerCache(tracerCreator);
+        ActivityTracer result1 = underTest.addEvent(activity, "beep1");
+        ActivityTracer result2 = underTest.addEvent(activity, "beep2");
+        ActivityTracer result3 = underTest.addEvent(activity, "beep3");
+        assertSame(activityTracer, result1);
+        assertSame(activityTracer, result2);
+        assertSame(activityTracer, result3);
+        verify(activityTracer).addEvent("beep1");
+        verify(activityTracer).addEvent("beep2");
+        verify(activityTracer).addEvent("beep3");
+        verifyNoMoreInteractions(tracerCreator);
+    }
+
+    @Test
+    void startSpanIfNoneInProgress() {
+        when(tracerCreator.apply(activity)).thenReturn(activityTracer);
+        when(activityTracer.startSpanIfNoneInProgress("wrenchy")).thenReturn(activityTracer);
+
+        ActivityTracerCache underTest = new ActivityTracerCache(tracerCreator);
+
+        ActivityTracer result = underTest.startSpanIfNoneInProgress(activity, "wrenchy");
+        assertSame(activityTracer, result);
+        verify(activityTracer).startSpanIfNoneInProgress("wrenchy");
+        verifyNoMoreInteractions(tracerCreator);
+    }
+
+    @Test
+    void initiateRestartSpanIfNecessary_singleActivity() {
+
+        when(tracerCreator.apply(activity)).thenReturn(activityTracer);
+        when(activityTracer.initiateRestartSpanIfNecessary(false)).thenReturn(activityTracer);
+
+        ActivityTracerCache underTest = new ActivityTracerCache(tracerCreator);
+
+        ActivityTracer result = underTest.initiateRestartSpanIfNecessary(activity);
+        assertSame(activityTracer, result);
+        verify(activityTracer).initiateRestartSpanIfNecessary(false);
+        verifyNoMoreInteractions(tracerCreator);
+    }
+
+    @Test
+    void initiateRestartSpanIfNecessary_multiActivity() {
+        Activity activity2 = new Activity() {
+                    // to get a new class name used in the cache
+                };
+        ActivityTracer activityTracer2 = mock(ActivityTracer.class);
+
+        when(tracerCreator.apply(activity)).thenReturn(activityTracer);
+        when(tracerCreator.apply(activity2)).thenReturn(activityTracer2);
+        when(activityTracer.addEvent(anyString())).thenReturn(activityTracer);
+        when(activityTracer.initiateRestartSpanIfNecessary(true)).thenReturn(activityTracer);
+
+        ActivityTracerCache underTest = new ActivityTracerCache(tracerCreator);
+
+        underTest.addEvent(activity, "foo");
+        underTest.addEvent(activity2, "bar");
```

Comment:
Any verification for this?

Reply:
No, because the event stuff is tested in other methods. This is purely state/setup code required to get the cache set up with more than 1 activity. The test readily fails without either of these calls.